### PR TITLE
feature(wallet-setup): hide mnemonic button until phrase is valid

### PR DESCRIPTION
**Pull request by banklesss**
Related to [YOMO-1325](https://emurgo.atlassian.net/browse/YOMO-1325) and [YOMO-1330](https://emurgo.atlassian.net/browse/YOMO-1330)

[YOMO-1325]: https://emurgo.atlassian.net/browse/YOMO-1325?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[YOMO-1330]: https://emurgo.atlassian.net/browse/YOMO-1330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/SetupWallet/common/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -88,6 +88,10 @@ export const TextInput = React.forwardRef((props: TextInputProps, ref: React.For
     </HelperText>
   )
 
+  React.useEffect(() => {
+    if (value === '') setIsWordValid(false)
+  }, [value])
+
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/common/TextInput/TextInput.tsx:94]: fix for CREAR ALL action

Comment (banklesss) [reply apps/wallet-mobile/src/features/SetupWallet/common/TextInput/TextInput.tsx:94]: it needs a big rebuild

**Review by stackchain — APPROVED** [review]

**Review by michalszorad — APPROVED** [review]

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx [hunk 2]
```diff
@@ -120,14 +120,16 @@ export const RestoreWalletScreen = () => {
...
-            disabled={isEmptyString(mnemonic)}
-            onPress={handleOnNext}
-          />
-        </View>
+        {mnemonic !== '' && (
+          <View style={styles.padding}>
+            <Button
+              title={strings.next}
+              style={styles.button}
+              disabled={isEmptyString(mnemonic)}
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx]: is it needed? since it won't show unless is <> '' ?

Comment (banklesss) [reply apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx]: i would left that there JIC 👀

Comment (banklesss) [reply apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx]: removed

**Comment by banklesss** [comment]
need to review one thing

**Comment by banklesss** [comment]
> need to review one thing

done

**Review by jorbuedo — APPROVED** [review]

**Pull request merged**